abad.go

```diff
@@ -0,0 +1,100 @@
+package abad
+
+import (
+	"fmt"
+
+	"github.com/NeowayLabs/abad/ast"
+	"github.com/NeowayLabs/abad/parser"
+	"github.com/NeowayLabs/abad/types"
+	"github.com/NeowayLabs/abad/token"
+)
+
+type (
+	// Obj types
+	Obj interface {
+		String() string
+	}
+
+	// Abad interpreter, a very bad one.
+	Abad struct {
+		filename string
+	}
+)
+
+func NewAbad(filename string) *Abad {
+	return &Abad{
+		filename: filename,
+	}
+}
+
+func (a *Abad) Eval(code string) (Obj, error) {
+	program, err := parser.Parse(a.filename, code)
+	if err != nil {
+		return nil, err
+	}
+
+	return a.eval(program)
+}
+
+func (a *Abad) eval(n ast.Node) (Obj, error) {
+	switch n.Type() {
+	case ast.NodeProgram:
+		return a.evalProgram(n.(*ast.Program))
+	case ast.NodeNumber:
+		val := n.(ast.Number)
+		return types.Number(val.Value()), nil
+	case ast.NodeUnaryExpr:
+		expr := n.(*ast.UnaryExpr)
+		return a.evalUnaryExpr(expr)
+	}
+
+	panic(fmt.Sprintf("AST(%s) not implemented", n))
+	return nil, nil
+}
+
+func (a *Abad) evalProgram(stmts *ast.Program) (Obj, error) {
```

Comment:
Hmm I see the evalProgram just goes recursively until it reaches numbers or unary expressions